### PR TITLE
Course-score-calculation Sentry error

### PR DESCRIPTION
**Pull request by sjagla27**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context
Sentry issue: https://sentry.io/organizations/ls1intum/issues/1983529117/?project=1440029&query=is%3Aunresolved&sort=freq&statsPeriod=14d

### Description
Check if the dueDate is undefined or null.

### Steps for Testing
1. Create a ProgrammingExercise and submit more than one solution.
2. Go to Course Overview and select the course the exercise is in.
3. Go to statistics and look for an error in the console.

### Test Coverage
unchanged

### CODE REVIEW

File: src/main/webapp/app/overview/course-score-calculation.service.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ export class CourseScoreCalculationService {
             });
 
             const gracePeriodInSeconds = 10;
-            if (dueDate === null || dueDate.add(gracePeriodInSeconds, 'seconds') >= resultsArray[0].completionDate!) {
+            if (!!!dueDate || dueDate.add(gracePeriodInSeconds, 'seconds') >= resultsArray[0].completionDate!) {
```
Comment (jpbernius) [inline src/main/webapp/app/overview/course-score-calculation.service.ts:153]: Would this be the same thing?
```suggestion
            if (dueDate == null || dueDate.add(gracePeriodInSeconds, 'seconds') >= resultsArray[0].completionDate!) {
```

Comment (sjagla27) [reply src/main/webapp/app/overview/course-score-calculation.service.ts:153]: This also checks for undefined

Comment (fde312) [reply src/main/webapp/app/overview/course-score-calculation.service.ts:153]: @jpbernius This should work as well. `==` checks for `null` and `undefined`.

Comment (jpbernius) [reply src/main/webapp/app/overview/course-score-calculation.service.ts:153]: I feel like `!!!` is less clear than `== null`.

Comment (jpbernius) [reply src/main/webapp/app/overview/course-score-calculation.service.ts:153]: @sjagla27 Note `==` vs `===`

Comment (sjagla27) [reply src/main/webapp/app/overview/course-score-calculation.service.ts:153]: Okay.

**Review by FuchsDominik — APPROVED** [review]
Code makes sense and just tested it the way you explained and no error occured in the console.

**Review by krusche — APPROVED** [review]

**Pull request merged**